waku/common/rate_limiter.go

```diff
@@ -228,30 +243,45 @@ func (r *PeerRateLimiter) Decorate(p RateLimiterPeer, rw p2p.MsgReadWriter, runL
 	return <-errC
 }
 
-// throttleIP throttles a number of messages incoming from a given IP.
-// It allows 10 packets per second.
-func (r *PeerRateLimiter) throttleIP(ip string) bool {
-	if r.LimitPerSecIP == 0 {
-		return false
-	}
+// throttleIP throttles packets incoming from a given IP.
+func (r *PeerRateLimiter) throttleIP(ip string, size uint32) bool {
 	if stringSliceContains(r.whitelistedIPs, ip) {
 		return false
 	}
-	return r.ipThrottler.Halt(ip, 1, r.LimitPerSecIP)
-}
 
-// throttlePeer throttles a number of messages incoming from a peer.
-// It allows 3 packets per second.
-func (r *PeerRateLimiter) throttlePeer(peerID []byte) bool {
-	if r.LimitPerSecIP == 0 {
-		return false
+	var packetLimiterResponse bool
+	var bytesLimiterResponse bool
+
+	if r.PacketLimitPerSecIP != 0 {
+		packetLimiterResponse = r.packetIPThrottler.Halt(ip, 1, r.PacketLimitPerSecIP)
 	}
+	if r.BytesLimitPerSecIP != 0 {
+		bytesLimiterResponse = r.bytesIPThrottler.Halt(ip, int64(size), r.BytesLimitPerSecIP)
+	}
+
+	return packetLimiterResponse || bytesLimiterResponse
+}
+
+// throttlePeer throttles packets incoming from a peer.
+func (r *PeerRateLimiter) throttlePeer(peerID []byte, size uint32) bool {
 	var id enode.ID
 	copy(id[:], peerID)
 	if enodeIDSliceContains(r.whitelistedPeerIDs, id) {
 		return false
 	}
-	return r.peerIDThrottler.Halt(id.String(), 1, r.LimitPerSecPeerID)
+
+	var packetLimiterResponse bool
+	var bytesLimiterResponse bool
+
+	if r.PacketLimitPerSecPeerID != 0 {
+		packetLimiterResponse = r.packetPeerIDThrottler.Halt(id.String(), 1, r.PacketLimitPerSecPeerID)
```

Comment:
Do we even need separate `Throttler` intances for IDs and IPs? They are different identifiers, they wouldn't clash.
I guess it makes the code a bit cleaner, and possible makes key searches shorter.

Reply:
Good idea